### ISSUE TITLE
How to merge label classes without modifying the ground truth data

### ISSUE THREAD

**Issue by antopost** [Deci-AI/super-gradients#795]
I want to take a set of model weights for semantic segmentation pre-trained on cityscapes and finetune it such that it ignores all classes other than road.
Is there currently a way to merge or ignore label classes, e.g. by passing an argument to the dataloader?
Else can you give a hint where best to modify the source code to achieve this?

My best guess would be to modify [this function](https://github.com/Deci-AI/super-gradients/blob/e2456ce30b93c2a06e41756bad8009d38b9f9444/src/super_gradients/training/datasets/segmentation_datasets/cityscape_segmentation.py#L138) to set other to be ignored.

**Comment by ofrimasad** [comment]
Hi @antopost .
Unfortunately, this feature is not available.
But as you suggested, a small change to the function you were pointing to will do the trick.
```
    @staticmethod
    def target_transform(target):
        """
        target_transform - Transforms the sample image
        This function overrides the original function from SegmentationDataSet and changes target pixels with value
        255 to value = CITYSCAPES_IGNORE_LABEL. This was done since current IoU metric from torchmetrics does not
        support such a high ignore label value (crashed on OOM)

            :param target: The target mask to transform
            :return:       The transformed target mask
        """
        out = SegmentationDataSet.target_transform(target)
        out[out == 255] = CITYSCAPES_IGNORE_LABEL

        # if 4, 6, 7, 8 are values you would like to ignore:
        out[out == 4] = CITYSCAPES_IGNORE_LABEL 
        out[out == 6] = CITYSCAPES_IGNORE_LABEL
        out[out == 7] = CITYSCAPES_IGNORE_LABEL
        out[out == 8] = CITYSCAPES_IGNORE_LABEL
        return out
```

**Comment by ofrimasad** [comment]
One thing you need to consider. 
Your data needs to have at least two classes (CITYSCAPES_IGNORE_LABEL is not considered a class).
if you remove ALL classes but the road, you may want to use the mapping instead of the CITYSCAPES_IGNORE_LABEL.
map all labels to one label (i.e. label 1) that you will consider to be the background
